src/Data/Factory.php

```diff
@@ -163,4 +163,14 @@ public function range(int $start, int $length) : Range
     {
         return new Range($start, $length);
     }
+
+    /**
+     * @param string $id
+     * @param mixed $direction Order::ASC|Order::DESC
+     * @return Order
```

Comment:
```suggestion
     * @param mixed $direction Order::ASC|Order::DESC
```

[We MAY drop doc-blocks if they do not add any information.](https://github.com/ILIAS-eLearning/ILIAS/blob/release_6/src/README.md) Also, this very much clarifies how doc-blocks could be a problem: The block talks about `$id`, while the parameters use the name `$subject`. If we don't duplicate the typehint the problem vanishes as well =)